Add the select and selectors modules

coolreader18 opened the pull request.

palaviv commented on vm/src/stdlib/select.rs:
```diff
@@ -0,0 +1,75 @@
+        Some(&mut x),
+        timeout.as_mut(),
+    )
+    .map_err(|err| super::os::convert_nix_error(vm, err))?;
```
We should check here if the select has failed due to an interrupt and if so we should check signals and continue. I would suggest just adding a `TODO` for now.

coolreader18 commented:
I've addressed your suggestions @palaviv; I didn't add tests because I wasn't quite sure how I would do them, maybe subprocesses on separate threads? What do you think?

palaviv commented:
Looks good. I would suggest using sockets for testing. I think that will be the easiest way.

coolreader18 commented:
I'm working on improving the `socket` module, I think I'll add tests based on that new version.
